### DISCUSSION TITLE
Choose language for embedded dashboard

### DISCUSSION THREAD

**Discussion by aehanno** [apache/superset#22536] [Q&A / Help]
How I can specified language when I display my embedded dashboard ?
Is it possible to specify it in the guest token or in the SDK method (supersetEmbeddedSdk.embedDashboard) ?

Thanks

**Comment by c2ofh** [comment]
@aehanno have you found any workaround yet?

**Reply by aehanno** [reply]
@c2ofh, No, I found anything
Do you have a need on this topic too ?

**Comment by lelelethu** [comment]
@aehanno I found this for 2 days. I am very interested in this issue.

**Comment by christophatinnuce** [comment]
I had no success setting the language via an API  call (Superset runs embedded in an iframe -> session mismatch).

This worked for me:

  CustomSecurityManager.py: |
    import jwt
    import json
    import sys
    import time
    from flask_appbuilder.security.views import AuthDBView
    from flask_appbuilder import expose
    from superset.security.manager import SupersetSecurityManager
    from flask import abort, current_app, flash, g, redirect, request, session, url_for
    from flask_login import login_user, logout_user
    class CustomAuthDBView(AuthDBView):
            login_template = 'appbuilder/general/security/login_db.html'
            @expose('/login/', methods=['GET', 'POST'])
            def login(self):
                    lang = request.args.get('lang')
                    if  lang:
                            print("Setting Lang from request: " + lang,  file=sys.stderr)
                            session["locale"] = lang
                            
You have to configure your CustomSecurityManager;

  superset_config.py  : |
    from xxx import CustomSecurityManager
    CUSTOM_SECURITY_MANAGER = CustomSecurityManager

**Reply by aehanno** [reply]
Oh I see, can be a good solution but on my side I didn't use OAuth to embedded my dashboard

**Reply by Pavel-Malai** [reply]
@christophatinnuce how did you send the lang parameter to the API?

**Comment by pavelmalai** [comment]
I had the same problem, implemented it by using a mutator:

Superset config file:


```
BABEL_DEFAULT_LOCALE = "en"
BABEL_DEFAULT_FOLDER = "superset/translations"
LANGUAGES = {
    "en": {"flag": "us", "name": "English"},
    "es": {"flag": "es", "name": "Spanish"}
}

def set_locale_from_url_param():
    from flask import session, request
    locale_param = request.args.get('lang')
    if locale_param:
        supported_locales = ['en', 'es']
        if locale_param in supported_locales:
            session["locale"] = locale_param

FLASK_APP_MUTATOR = lambda app: app.before_request(set_locale_from_url_param)
```


And when emedding the dashboard, add this url param to the dashboardUIConfig

```
dashboardUiConfig: {
     urlParams:{
          lang: 'es'
}
```

**Reply by jacozzola** [reply]
Not working for me.
When trying this method superset logs the following error
`ERROR:superset.translations.utils:Error loading language pack for, falling back on en`
